src/v2/models/AppState.js

```diff
@@ -103,9 +103,10 @@ export default Model.extend({
   setIonResponse (resp) {
     const idx = this.get('idx');
     // Don't re-render view if new response is same as last.
-    // Usually happening at polling and pipeline doesn't proceed to next step.
-    // expiresAt will be different for each response, hence compare objects without that property
-    if (_.isEqual(_.omit(idx.rawIdxState, 'expiresAt'), _.omit(this.get('__previousResponse'), 'expiresAt'))) {
+    // if stateHandle changes re-render view.
+    // stateHandle does not change for instrospect/poll calls
+    if (_.omit(idx.rawIdxState, 'expiresAt').stateHandle ===
```

Comment:
can't we do ?
`if (idx.rawIdxState.stateHandle === this.get('__previousResponse').stateHandle)`